### ISSUE TITLE
Is there a way to clip the shimmer to the child below?

### ISSUE THREAD

**Issue by dheerajbred** [maddyb99/shimmer_examples#2]
Not an issue.
Is there a way to clip the shimmer to the child below? The shimmer is overflowing. Like a Cupertinobutton with shimmer

**Comment by maddyb99** [comment]
I understand this issue is related to the widget itself. If that is the case, I would like to request you to open this issue in the maddyb99/shimmer_animation repository.
This repository just maintains sample implementations of skeleton loaders using ```shimmer_animation``` . If you are requesting a sample, I would be happy to help you with that also!

Also, I would highly appreciate some illustrations of what actually happens and what you would like to achieve! This is just so we are on the same page!

Thank you for showing interest in this🙏🙏

**Comment by maddyb99** [comment]
I'm assuming what you want to do is tightly wrap the shimmer animation around the child widget i.e., in this, case a CupertinoButton and I'm assuming that by overflow you mean that the Shimmer is slightly bigger than the child itself, am I right?

The Shimmer widget is designed to tightly wrap around the child widget and actually does that pretty well.
The only exception to this might seem to be when the child widget has a custom shape/border radius. But consider this, if we have a container wrapped in a card, changing the border radius of the container won't affect the card.

And I'm also assuming that this is the problem you are facing.

So, the two possible solutions are:
1. Updating the Shimmer widget to accept a shape like a button would. This has been on my radar for quite some time now but I've been a bit busy and no one raised such an issue😅 
2. The quicker solution, right now, would be to wrap Shimmer widget with a [DecoratedBox](https://api.flutter.dev/flutter/widgets/DecoratedBox-class.html) or a Container and give them a shape. Another way would be to wrap the shimmer widget with a clipper for a custom path.

I hope this clears up any of the doubts that you have!!

**Comment by maddyb99** [comment]
Also, I would update the widget to accept a shape after my exams. So you could expect an update by the end of January!

**Comment by dheerajbred** [comment]
Yes, I am referring to child widgets with different boundary like rounded cornered button or a variable shape or an icon perhaps. 

I also tried using https://pub.dev/packages/shimmer This clips the boundaries correctly but it acts as a shader overlay i.e the child rendering will get completely overlayed with this. 

You package on the other hand just stacks the shimmer on top on the child, just the way I want. 
I am trying to tweek shimmer package for this need. I'll see how it goes....I am a newbie though😅

Not a problem. No hurries or worries. This is just a request and any update to it is appreciated a lot👍

**Comment by maddyb99** [comment]
I really like the [shimmer](https://pub.dev/packages/shimmer) package but I find it to be a bit complicated for someone new.
So, I built this package just to simplify the implementation of a Shimmer animation and give developers something ready to ship out of the box while providing options to tweak.

An example for implementing a rounded rectangle would be:
```dart
ClipRRect(
      borderRadius: BorderRadius.all(Radius.circular(50)),
      child: Shimmer(
        enabled: isDisabled,
        child: CupertinoButton(
          borderRadius: BorderRadius.all(Radius.circular(50)),
          color: Colors.blue[900],
          disabledColor: Colors.grey,
          child: Text('Login'),
          onPressed: onPressed,
        ),
      ),
   )
``` 
This is what this would look like:

<img  height="400" src="https://user-images.githubusercontent.com/43785086/104633209-e1fd8880-56c4-11eb-86da-7706c326a4f3.gif">

I've ommitted onPressed for clarity.

For custom shapes we could use:

- **Rectangle:** [ClipRect](https://api.flutter.dev/flutter/widgets/ClipRect-class.html)
- **Oval:** [ClipOval](https://api.flutter.dev/flutter/widgets/ClipOval-class.html)
- **Any other shape:** [ClipPath](https://api.flutter.dev/flutter/widgets/ClipPath-class.html)

**Comment by maddyb99** [comment]
@dheerajbred please let me know if your issue is resolved for now!!
I will now transfer this issue to the main repo and start working on this soon!
Cheers :clinking_glasses:

**Comment by dheerajbred** [comment]
Yes, ClipRRect definitely solves the issue for rounded cornered widgets👍
